File: frontend/types/trip.ts
export type Trip = {
  id: number;
  destination: string;
  budget: number;
  days: number;
  category: string;
  daily_budget: number;
};

export type TripInput = {
  destination: string;
  budget: number;
  days: number;
  travel_style?: string;
};

File: backend/models/trip.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.sql import func

try:
    from ..database import Base
except ImportError:
    from database import Base


class Trip(Base):
    __tablename__ = "trips"
    
    id = Column(Integer, primary_key=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    destination = Column(String, nullable=False)
    days = Column(Integer, nullable=False)
    budget = Column(Float, nullable=False)
    category = Column(String, nullable=False)
    daily_budget = Column(Float, nullable=False)